cpp/src/arrow/compute/kernels/vector_sort_test.cc

```diff
@@ -368,6 +368,138 @@ TEST(ArraySortIndicesFunction, Array) {
   AssertDatumsEqual(expected, actual, /*verbose=*/true);
 }
 
+TEST(ArraySortIndicesFunction, DictionaryArray) {
```

Comment:
Perhaps add a test with a non-zero array indices and/or values offset somewhere?